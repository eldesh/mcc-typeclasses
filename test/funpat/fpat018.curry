-- Since function patterns effectively get transformed into additional
-- guard expressions in the scope of the function pattern, nested
-- patterns within the arguments of a function pattern cannot be used to
-- dynamically introduce type class instances for their own
-- arguments. Instead, those instances must be provided by the context.

data T a = Eq a => C a
f (C c) = c
g (f (C c)) | c == c = c
g' fp = case fp of f (C c) | c == c -> c
