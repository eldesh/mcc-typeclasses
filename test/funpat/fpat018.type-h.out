Eq _1 => T _1 -> _1
